src/containers/classrooms/ClassroomsTableContainer.jsx

```diff
@@ -1,82 +1,91 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Actions } from 'jumpstate';
-import AstroClassroomsTable from '../../components/astro/AstroClassroomsTable';
+import Paragraph from 'grommet/components/Paragraph';
 
+import ConfirmationDialog from '../../components/common/ConfirmationDialog';
+import DarienClassroomsTable from '../../components/darien/DarienClassroomsTable';
+import AstroClassroomsTableContainer from '../astro/AstroClassroomsTableContainer';
 import {
   CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
 } from '../../ducks/classrooms';
 import {
   ASSIGNMENTS_INITIAL_STATE, ASSIGNMENTS_PROPTYPES
 } from '../../ducks/assignments';
-import { programsMocks } from '../../ducks/programs';
 
 class ClassroomsTableContainer extends React.Component {
   constructor() {
     super();
 
     this.state = {
-      toExport: {
-        assignment: {},
-        classroom: {}
-      }
+      classroomToDelete: null,
+      showConfirmationDialog: false
     };
 
-    this.onExportModalClose = this.onExportModalClose.bind(this);
-    this.showExportModal = this.showExportModal.bind(this);
+    this.closeConfirmationDialog = this.closeConfirmationDialog.bind(this);
+    this.deleteClassroom = this.deleteClassroom.bind(this);
+    this.maybeDeleteClassroom = this.maybeDeleteClassroom.bind(this);
   }
 
-  onExportModalClose() {
-    this.setState({ toExport: { assignment: {}, classroom: {} } });
+  maybeDeleteClassroom(id) {
+    this.setState({ classroomToDelete: id, showConfirmationDialog: true });
+  }
 
-    Actions.caesarExports.showModal();
+  closeConfirmationDialog() {
+    this.setState({ classroomToDelete: null, showConfirmationDialog: false });
   }
 
-  showExportModal(assignment, classroom) {
-    this.setState({ toExport: { assignment, classroom } });
+  deleteClassroom() {
+    if (this.state.classroomToDelete === null) return;
+
+    Actions.deleteClassroom(this.state.classroomToDelete).then((response) => {
+      // TODO: For API optimization, do we want to instead manually remove the classroom
+      // out of local app state instead of making another API call
+      Actions.getClassroomsAndAssignments();
+      this.closeConfirmationDialog();
 
-    Actions.caesarExports.showModal();
-    Actions.getCaesarExport({ assignment, classroom });
+      if (response) {
+        Actions.classrooms.setToastState({ status: 'ok', message: 'Classroom deleted' });
+      }
+    });
   }
 
-  selectClassroom(classroom) {
-    Actions.classrooms.selectClassroom(classroom);
+  determineClassroomsTable() {
+    const ClassroomsTable = this.props.selectedProgram.custom ? DarienClassroomsTable : AstroClassroomsTableContainer;
```

Comment:
OK, now I see what you meant when you said _"...after puzzling on this for a while I ended up settling with a ternary"_

I think this is the way to go, at least for now. Re: _"I'd really like to try to clean this up at some point",_ I'm all for it, but I'm OK to launch something that works first and refactor later. That said...

Thought: I imagine the ideal solution would be to properly plan out a kind of "generic Education Program" system similar to Panoptes' "generic Build-Your-Own Project" system, then plunk the switching code into a central "ProgramsController"/"ProgramsConfigToComponentTranslator" thingy...

```
switch (this.props.selectedProgram.settings.programType) {
  case "wildlife_camera_trap_classroom": ClassroomsTable = DarienClassroomTable; break;
  case "standard_auto_created_assignments_classroom": ClassroomsTable = AstroClassroomsTableContainer; break;
  ...
}
return ClassroomsTable;
```

...but that _might_ be a pretty darn deep rabbit hole to go down.

Reply:
It's a good idea, but not worth the time right now even though I dislike what I ended up doing! If another curricula/educational program is added after I2A and the Wildcams, then that's when this should be refactored.

Reply:
👍